dart-mode.el

```diff
@@ -562,6 +582,298 @@ This uses `dart-format-path' to find the formatter."
       (dart-format-region beg (point)))))
 
 
+;;; Dart analysis server
+
+(cl-defstruct
+    (dart--analysis-server
+     (:constructor dart--make-analysis-server))
+  "Struct containing data for an instance of a Dart analysis server.
+
+The slots are:
+- `process': the process of the running server.
+- `buffer': the buffer where responses from the server are written."
+  process buffer)
+
+(defvar dart-debug nil
+  "If non-nil, enables writing debug messages for dart-mode.")
+(defcustom dart-enable-analysis-server nil
```

Comment:
Nit: There should be a newline between all of these.

The defcustoms should have more metadata. For example, this should have:

```elisp
  :type 'boolean
  :package-version '(dart-mode . "0.12")
  :group dart-mode
```

See [the docs on customization](http://www.gnu.org/software/emacs/manual/html_node/elisp/Customization.html) for details. You'll also need to `defgroup` a group for them to be in.